### PR TITLE
CI: allow for self-approvals for maintainers

### PR DESCRIPTION
**Pull request by keradus**
as aligned between maintainers

### CODE REVIEW

File: .github/workflows/self-approval.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        id: marker
+        run: |
+          msg=$(git show -s --format=%s ${{ github.event.after }})
+          echo $msg
+          indicator=$([[ $msg == $MARKER ]] && echo yes || echo no)
+          echo $indicator
+          echo "indicator_value=${indicator}" >> "$GITHUB_OUTPUT"
+      - name: skip
+        if: steps.marker.outputs.indicator_value != 'yes'
+        run: echo no marker detected, skipping
```
Comment (Wirone) [inline .github/workflows/self-approval.yml:37]: I am not sure we need this. Personally I would keep only triggered path, if `indicator_value` is not `yes`, then the next step will be skipped and it is enough to see that conditions were not met.

```suggestion
```

File: .github/workflows/self-approval.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      - name: self-approve
```
Comment (Wirone) [inline .github/workflows/self-approval.yml:38]:
```suggestion
      - name: Approve and enable auto-merge
```

**Review by Wirone — CHANGES_REQUESTED** [review]

File: .github/workflows/self-approval.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+on:
+  - pull_request_target
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  job:
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (keradus) [reply .github/workflows/self-approval.yml]: happy to unify. maybe some linter for future, to not have this unified manually :D (yeah, likely overkill)

File: .github/workflows/self-approval.yml (hunk 1 above)
Comment (keradus) [reply .github/workflows/self-approval.yml:38]: yeah, i used it for debugging mostly :D

File: .github/workflows/self-approval.yml (hunk 2 above)
Comment (keradus) [reply .github/workflows/self-approval.yml:36]: soooo loooong :D

File: .github/workflows/self-approval.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# Flow for self-approving the PRs, used by maintainers for cases aligned between them.
+# To trigger it, maintainer need to commit and push:
```
Comment (keradus) [inline .github/workflows/self-approval.yml:2]:
```suggestion
# To trigger it, the maintainer needs to commit and push:
```

File: .github/workflows/self-approval.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  automate:
+    name: Automate
+    runs-on: ubuntu-latest
+    if: ${{ contains(fromJson('["julienfalque", "keradus", "keradus-ci", "kubawerlos", "localheinz", "Wirone"]'), github.actor) }}
```
Comment (Wirone) [inline .github/workflows/self-approval.yml:18]: Why it wasn't included in this PR's CI? You're the actor, so it should be triggered?

Comment (Wirone) [reply .github/workflows/self-approval.yml:18]: Ah, right. So let's merge 🙂.

**Review by Wirone — APPROVED** [review]

**Pull request merged**